File: all-submissions/0189-rotate-array/solution.java
class Solution {
    public void rotate(int[] nums, int k) {

        int[] ans = new int[nums.length];
        if (k%nums.length == 0) return;
        k = k%nums.length;
        int startIndex = nums.length - k;
        int j=0;
        for(int i=startIndex;i<nums.length;i++) {
            ans[j++] = nums[i];
        }
        for(int i=0; i<startIndex; i++){
            ans[j++] = nums[i];
        }
        for(int i=0; i<nums.length; i++){
            nums[i] = ans[i];
        }
        return;
        
    }
}

File: all-submissions/0189-rotate-array/solution.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        n = len(nums)
        steps = k%n
        if steps == 0:
            return
        def reverse(start, end):
            while start<end:
                nums[start], nums[end] = nums[end], nums[start]
                start, end = start+1, end-1

        reverse(0, n-1)
        reverse(0, steps-1)
        reverse(steps, n-1)
        return
